sdk/keyvault/keyvault-admin/src/accessControlClient.ts

```diff
@@ -0,0 +1,442 @@
+        yield currentSetResponse.value.map(mappings.roleAssignment.generatedToPublic, this);
+      } else {
+        break;
+      }
+    }
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the iteration of all the available results of {@link listRoleAssignments}.
+   * @param {string} scope The scope of the role assignments.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleAssignmentsAll(
+    scope: RoleAssignmentScope,
+    options?: ListRoleAssignmentsOptions
+  ): AsyncIterableIterator<KeyVaultRoleAssignment> {
+    const f = {};
+
+    for await (const page of this.listRoleAssignmentsPage(scope, f, options)) {
+      for (const item of page) {
+        yield item;
+      }
+    }
+  }
+
+  /**
+   * Iterates over all of the available role assignments in an Azure Key Vault.
+   *
+   * Example usage:
+   * ```ts
+   * let client = new AccessControlClient(url, credentials);
+   * for await (const roleAssignment of client.listRoleAssignments("/")) {
+   *   console.log("Role assignment: ", roleAssignment);
+   * }
+   * ```
+   * @summary Lists all of the role assignments in a given scope.
+   * @param {string} scope The scope of the role assignments.
+   * @param {ListPropertiesOfKeysOptions} [options] The optional parameters.
+   */
+  public listRoleAssignments(
+    scope: RoleAssignmentScope,
+    options: ListRoleAssignmentsOptions = {}
+  ): PagedAsyncIterableIterator<KeyVaultRoleAssignment> {
+    const requestOptions = operationOptionsToRequestOptionsBase(options);
+    const span = createSpan("listRoleAssignments", requestOptions);
+    const updatedOptions: ListRoleAssignmentsOptions = {
+      ...requestOptions,
+      ...setParentSpan(span, requestOptions)
+    };
+
+    const iter = this.listRoleAssignmentsAll(scope, updatedOptions);
+
+    span.end();
+    return {
+      next() {
+        return iter.next();
+      },
+      [Symbol.asyncIterator]() {
+        return this;
+      },
+      byPage: (settings: PageSettings = {}) =>
+        this.listRoleAssignmentsPage(scope, settings, updatedOptions)
+    };
+  }
+
+  /**
+   * @internal
+   * @ignore
+   * Deals with the pagination of {@link listRoleDefinitions}.
+   * @param {string} scope The scope of the role definition.
+   * @param {PageSettings} continuationState An object that indicates the position of the paginated request.
+   * @param {ListRoleAssignmentsOptions} [options] Common options for the iterative endpoints.
+   */
+  private async *listRoleDefinitionsPage(
+    scope: RoleAssignmentScope,
+    continuationState: PageSettings,
+    options?: ListRoleDefinitionsOptions
+  ): AsyncIterableIterator<KeyVaultRoleDefinition[]> {
+    if (continuationState.continuationToken == null) {
+      const optionsComplete: RoleAssignmentsListForScopeOptionalParams = {
+        // Not supported!
+        // maxresults: continuationState.maxPageSize,
```

Comment:
Did you intend to leave this in?

Reply:
I left it intentionally to bring attention to it. Since our other clients for Key Vault do support this, I wonder if we should leave it there for people looking at the source? Let me know if you don't think it can be valuable and I'll remove it.

Reply:
This gets resolved by my fix to https://github.com/Azure/azure-sdk-for-js/pull/10815/files/6ab4ae11154ef9b118002bed7def865cf7fd0226#r481471012